loadbalancer/algorithm.go

```diff
@@ -0,0 +1,118 @@
+package loadbalancer
+
+import (
+	"errors"
+	"math/rand"
+	"net/url"
+	"sync"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/zalando/skipper/eskip"
+	"github.com/zalando/skipper/routing"
+)
+
+type algorithmType int
+
+const (
+	none algorithmType = iota
+	roundRobinAlgorithm
+)
+
+type roundRobin struct {
+	mx    sync.Mutex
+	index int
+}
+
+type algorithmProvider struct{}
+
+var algorithms = map[algorithmType]func(int) routing.LBAlgorithm{
```

Comment:
I think I don't like the idea of this signature, because it binds it to simple algorithms only (roundrobin and random are fine but everything that needs to lookup metrics data for an endpoint would be not).